base-template.sh

```diff
@@ -484,27 +484,76 @@ run_hook_file() {
 #   1 in case a hook fails, 0 otherwise
 #####################################################
 process_shared_hooks() {
-    SHARED_REPOS_LIST="$1"
-    shift
+    SHARED_HOOKS_TYPE="$1"
+    SHARED_REPOS_LIST="$2"
+    shift 2
 
     update_shared_hooks_if_appropriate "$@"
     execute_shared_hooks "$@" || return 1
 }
 
 #####################################################
-# Sets the SHARED_ROOT and NORMALIZED_NAME
-#   for the shared hook repo url `$1`.
+# Check if `$1` is not a supported git clone url and
+#   is treated as a local path to a repository
+# Returns:
+#   none
+#####################################################
+is_local_url() {
+    if echo "$1" | grep -Eqv "[a-zA-Z]+://"; then
+        return 0
+    fi
+    return 1
+}
+
+#####################################################
+# Sets the `SHARED_ROOT` for the shared hook repo
+#    url `$1` and sets
+#   `SHARED_REPO_IS_CLONED` to `true` and its
+#   `SHARED_REPO_CLONE_URL` if is needs to get
+#    cloned and `SHARED_REPO_IS_LOCAL` to `true`
+#    if `$1` points to to a local path.
 #
 # Returns:
 #   none
 #####################################################
 set_shared_root() {
-    NORMALIZED_NAME=$(echo "$1" |
-        sed -E "s#.*[:/](.+/.+)\\.git#\\1#" |
-        sed -E "s/[^a-zA-Z0-9]/_/g")
-    SHARED_ROOT="$INSTALL_DIR/shared/$NORMALIZED_NAME"
-}
 
+    SHARED_REPO_CLONE_URL=""
+    SHARED_REPO_CLONE_BRANCH=""
+    SHARED_REPO_IS_LOCAL="false"
+    SHARED_REPO_IS_CLONED="true"
+
+    # Split "...@(.*)"
+    if echo "$1" | grep -q "@"; then
+        URL="$(echo "$1" | sed -E "s|^(.*)@.*$|\\1|")"
+        BRANCH="$(echo "$1" | sed -E "s|^.*@(.*)$|\\1|")"
+    else
+        URL="$1"
+        BRANCH="master"
```

Comment:
I think this would be whatever you set as default branch on the server, right?

Reply:
ah, ok, so we set it to empty and then dont hand that parameter to the clone function